### PR TITLE
feat: E2EE room setup header

### PR DESCRIPTION
**Pull request by yash-rajpal**
This is a new Header for the E2EE room setup flow, idea is that we shouldn't show all the room actions on the setup window and only a few items makes sense to be kept on this Header.

This PR introduces a new E2EERoomHeaderToolbox which is a toolbox for this specific use-case which contains only 3 room actions, e2e, room-info and members tab.

## Further comments

[E2EE-27](https://rocketchat.atlassian.net/browse/E2EE2-27)

[E2EE-27]: https://rocketchat.atlassian.net/browse/E2EE-27?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hugocostadev — DISMISSED** [review]

File: apps/meteor/client/views/room/Header/Header.tsx [hunk 1]
```diff
@@ -46,7 +46,15 @@ const Header = ({ room }: HeaderProps<IRoom>): ReactElement | null => {
 		return <VoipRoomHeader slots={slots} room={room} />;
 	}
 
-	return <RoomHeader slots={slots} room={room} topic={room.topic} />;
+	if (shouldDisplayE2EESetup) {
+		return <RoomHeaderE2EESetup room={room} topic={room.topic} slots={slots} />;
+	}
+
+	if (room.t === 'd' && (room.uids?.length ?? 0) < 3) {
```
Comment (MarcosSpessatto) [inline apps/meteor/client/views/room/Header/Header.tsx:53]: There is a helper for this check `isDirectMessageRoom`: https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/core-typings/src/IRoom.ts#L139

Also, what uids being greater than 3 mean? Would it be better to have this in a named variable maybe?

File: apps/meteor/client/views/room/Header/RoomHeaderE2EESetup.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const RoomHeaderE2EESetup = ({ room, topic = '', slots = {} }: RoomHeaderProps) => {
+	const e2eeState = useE2EEState();
+	const e2eRoomState = useE2EERoomState(room._id);
+
+	if (e2eeState === E2EEState.SAVE_PASSWORD || e2eeState === E2EEState.ENTER_PASSWORD || e2eRoomState === E2ERoomState.WAITING_KEYS) {
+		return <RoomHeader room={room} topic={topic} slots={slots} roomToolbox={<RoomToolboxE2EESetup />} />;
+	}
+
+	if (room.t === 'd' && (room.uids?.length ?? 0) < 3) {
```
Comment (MarcosSpessatto) [inline apps/meteor/client/views/room/Header/RoomHeaderE2EESetup.tsx:21]: Let's use the helper here: https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/core-typings/src/IRoom.ts#L139

Also, what uids being greater than 3 mean? Would it be better to have this in a named variable maybe?

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: .changeset/gold-flowers-shake.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@rocket.chat/i18n': patch
+'@rocket.chat/meteor': patch
```
Comment (MarcosSpessatto) [inline .changeset/gold-flowers-shake.md:3]:
```suggestion
'@rocket.chat/i18n': minor
'@rocket.chat/meteor': minor
```

File: apps/meteor/client/views/room/Header/Header.tsx [hunk 4]
```diff
@@ -18,6 +19,9 @@ type HeaderProps<T> = {
 
 const Header = ({ room }: HeaderProps<IRoom>): ReactElement | null => {
 	const { isMobile, isEmbedded, showTopNavbarEmbeddedLayout } = useLayout();
+	const encrypted = Boolean(room.encrypted);
+	const unencryptedMessagesAllowed = useSetting('E2E_Allow_Unencrypted_Messages');
```
Comment (MarcosSpessatto) [inline apps/meteor/client/views/room/Header/Header.tsx:23]:
```suggestion
	const unencryptedMessagesAllowed = useSetting<boolean>('E2E_Allow_Unencrypted_Messages');
```

**Review by MarcosSpessatto — APPROVED** [review]

**Review by hugocostadev — APPROVED** [review]

**Pull request merged**